app/views/components/_navigation-sidebar.html.erb

```diff
@@ -0,0 +1,50 @@
+<%
+  related_items ||= []
+  topics ||= []
+  publishers ||= []
+  collections ||= []
+  policies ||= []
+  external_links ||= []
+  related_content = [
+    {"related_items" => related_items},
+    {"topics" => topics},
+    {"publishers" => publishers},
+    {"collections" => collections},
+    {"policies" => policies},
+    {"external_links" => external_links}
+  ]
+%>
+<% if related_content.map(&:values).flatten.any? %>
+  <aside class="app-c-navigation-sidebar" role="complementary">
+    <% related_content.each do |section| %>
+      <% section.each do |section_title, links| %>
+
+        <% if section_title === "related_items" %>
+          <h2 id="nav-sidebar-<%= section_title %>" class="app-c-navigation-sidebar__main-heading" ><%= t('components.navigation_sidebar.related_content') %></h2>
+          <nav role="navigation" class="app-c-navigation-sidebar__nav-section" aria-labelledby="nav-sidebar-<%= section_title %>" %>
+            <ul>
+              <% links.each do |link| %>
+              <li class="app-c-navigation-sidebar__link">
+                <%= sidebar_link_to(link, section_title) %>
+              </li>
+              <% end %>
+            </ul>
+          </nav>
+
+        <% elsif links.any? %>
+          <h3 id="nav-sidebar-<%= section_title %>" class="app-c-navigation-sidebar__sub-heading <%= 'app-c-navigation-sidebar__external-heading' if section_title == "external_links" %>"><%= t('components.navigation_sidebar.' + section_title) %></h3>
```

Comment:
Nitpick: this could be broken into more than one line to make it more readable. Same goes for the H2 above.